proof/refine/ARM/InvariantUpdates_H.thy

```diff
@@ -90,15 +201,43 @@ lemma inQ_context[simp]:
 
 lemma valid_tcb'_tcbQueued[simp]:
   "valid_tcb' (tcbQueued_update f tcb) = valid_tcb' tcb"
-  by (cases tcb, rule ext, simp add: valid_tcb'_def tcb_cte_cases_def cteSizeBits_def)
+  by (cases tcb, rule ext, simp add: valid_tcb'_def tcb_cte_cases_def tcb_cte_cases_neqs)
 
 lemma valid_tcb'_tcbFault_update[simp]:
   "valid_tcb' tcb s \<Longrightarrow> valid_tcb' (tcbFault_update f tcb) s"
-  by (clarsimp simp: valid_tcb'_def  tcb_cte_cases_def cteSizeBits_def)
+  by (clarsimp simp: valid_tcb'_def  tcb_cte_cases_def tcb_cte_cases_neqs)
 
 lemma valid_tcb'_tcbTimeSlice_update[simp]:
   "valid_tcb' (tcbTimeSlice_update f tcb) s = valid_tcb' tcb s"
-  by (simp add:valid_tcb'_def tcb_cte_cases_def cteSizeBits_def)
+  by (simp add:valid_tcb'_def tcb_cte_cases_def tcb_cte_cases_neqs)
+
+lemma invs'_wu[simp]:
+  "invs' (ksWorkUnitsCompleted_update f s) = invs' s"
+  apply (simp add: invs'_def cur_tcb'_def valid_state'_def valid_bitmaps_def
+                   valid_irq_node'_def valid_machine_state'_def
+                   ct_not_inQ_def ct_idle_or_in_cur_domain'_def tcb_in_cur_domain'_def
+                   bitmapQ_defs)
+  done
+
+(* FIXME arch-split: MOVE *)
+context Arch begin arch_global_naming
+
+lemma valid_arch_state'_interrupt[simp]:
+  "valid_arch_state' (ksInterruptState_update f s) = valid_arch_state' s"
+  by (simp add: valid_arch_state'_def)
+
+end
+
+(* FIXME arch-split: locales? *)
+arch_requalify_facts
+  valid_arch_state'_interrupt (* FIXME arch-split: safe to export? any arch where this wouldn't be true *)
```

Comment:
If it's true between AARCH64/ARM_HYP and ARM then it should be true everywhere. The only difference in interrupt state handling would be in the hyp code.

Reply:
Cheers. Will remove the safety question comments